File: Backend/HrmApp.Api/HrmApp.Api/HrmDTO/EmployeeDTO.cs
using HrmApp.Api.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HrmApp.Api.HrmDTO
{
    public class EmployeeDTO
    {
        public int IdClient { get; set; }

        [Key]
        public int Id { get; set; }

        [StringLength(250)]
        public string? EmployeeName { get; set; }

        [StringLength(250)]
        public string? EmployeeNameBangla { get; set; }

        public byte[]? EmployeeImage { get; set; }

        [StringLength(250)]
        public string? FatherName { get; set; }

        [StringLength(250)]
        public string? MotherName { get; set; }

        
        public int? IdReportingManager { get; set; }

        public int? IdJobType { get; set; }

        public int? IdEmployeeType { get; set; }

        public DateTime? BirthDate { get; set; }

        public DateTime? JoiningDate { get; set; }


        public int? IdGender { get; set; }

        public int? IdReligion { get; set; }

        public int IdDepartment { get; set; }

        public int IdSection { get; set; }

        public int? IdDesignation { get; set; }

        public bool? HasOvertime { get; set; }

        public bool? HasAttendenceBonus { get; set; }

        public int? IdWeekOff { get; set; }

        [StringLength(250)]
        public string? Address { get; set; }

        [StringLength(250)]
        public string? PresentAddress { get; set; }

        [StringLength(30)]
        public string? NationalIdentificationNumber { get; set; }

        [StringLength(250)]
        public string? ContactNo { get; set; }

        public int? IdMaritalStatus { get; set; }

        public bool? IsActive { get; set; }

        public DateTime? SetDate { get; set; }

        [StringLength(50)]
        public string? CreatedBy { get; set; }



        public List<DocummentDto> EmployeeDocuments { get; set; } = [];

        public List<EducationInfoDto> EmployeeEducationInfos { get; set; } = [];
        public List<EmployeefamilyInfoDto> EmployeeFamilyInfos { get; set; } = [];
        public List<EmployeeProfessionalCertificationDto> EmployeeProfessionalCertifications { get; set; } = [];

        public IFormFile? ProfileFile { get; set; }
        public string? EmployeeImageBase { get; internal set; }
    }
}

File: Frontend/HRM_App_Frontend/src/app/HRM_Models/employeeDto.ts
import { DocummentDto } from "./documentDto";
import { EducationInfoDto } from "./educationInfoDto";
import { EmployeefamilyInfoDto } from "./employeefamilyInfoDto";
import { EmployeeProfessionalCertificationDto } from "./employeeProfessionalCertificationDto";

export interface EmployeeDTO{
    
    id: number;
    idClient: number;
    employeeName: string;
    employeeNameBangla: string;
    fatherName: string;
    motherName: string;
    birthDate: Date;
    joiningDate: Date;
    idDepartment: number;
    departmentName: string,
    idSection: number,
    sectionName: string,
    idDesignation: number,
    designation: string,
    address: string,
    idGender: number,
    genderName: string,
    idReligion?: number;
    religionName?: string;
    idReportingManager?: number;
    reportingManager?: string;
    idJobType?: number;
    jobTypeName?: string;
    idEmployeeType?: number;
    typeName?: string;
    presentAddress?: string;
    nationalIdentificationNumber?: string;
    contactNo?: string;
    isActive?: boolean;
    hasOvertime?: boolean;
    hasAttendenceBonus?: boolean;
    idWeekOff?: number;
    weekOffDay?: string;
    idMaritalStatus?: number;
    maritalStatusName?: string;
    setDate?: Date;
    createdBy?: string;
    profileFile?: File;
    employeeImageBase?: string;
    EmployeeDocuments: DocummentDto[];
    EmployeeEducationInfos: EducationInfoDto[];
    EmployeeProfessionalCertifications: EmployeeProfessionalCertificationDto[];
    EmployeeFamilyInfos: EmployeefamilyInfoDto[];
    

}


